### PR TITLE
Python3 transition

### PR DESCRIPTION
**Pull request by aali1987**

### CODE REVIEW

**Comment by aali1987** [comment]
@shadowwalkersb   We are going to soon merge this branch to master branch for our python3 release. 
Somehow travis and jenkin in mac is failing the build. Can you please have a look into why it is failing .

**Comment by shadowwalkersb** [comment]
Does this branch work for you locally on Mac?

**Comment by shadowwalkersb** [comment]
First failing commit might be https://github.com/cryoem/eman2/pull/465/commits/3a6c9452ad0dd95028faa68a017d01373fa8a058.

**Comment by aali1987** [comment]
@shadowwalkersb  You are right. It started to fail when we brought back the function  register_numpy_to_emdata as you can see in 3a6c945.
Is it possible that bringing the function back may have caused some compatibility issues?
But it works fine in Linux as we have tested it.

File: libpyEM/libpyTypeConverter2.cpp [hunk 1]
```diff
@@ -44,6 +44,10 @@
 using namespace boost::python;
 namespace np = boost::python::numpy;
 
+int init_numpy()
+{
+    import_array();
+}
```
Comment (shadowwalkersb) [inline libpyEM/libpyTypeConverter2.cpp:50]: This function expects `return`.

Comment (aali1987) [reply libpyEM/libpyTypeConverter2.cpp]: Yeah that should not be like this. Thanks for identifying the problem

**Comment by aali1987** [comment]
@shadowwalkersb   thanks for identifying the problem. Now all the tests pass successfully. i

File: libpyEM/libpyTypeConverter2.cpp [hunk 2]
```diff
@@ -62,7 +63,8 @@ BOOST_PYTHON_MODULE(libpyTypeConverter2)
         .staticmethod("numpy2em")
     ;
 
-    init_numpy();
+    int dummy = 0;
+    dummy = init_numpy();
```
Comment (shadowwalkersb) [inline libpyEM/libpyTypeConverter2.cpp]: Is this necessary?

Comment (aali1987) [reply libpyEM/libpyTypeConverter2.cpp]: if we follow the old method like we use to do in py2 then we get an error of "ISO C++ forbids declaration of init_numpy with no type [-fpermissive] .

For the time being lets keep it like this :)

Comment (shadowwalkersb) [reply libpyEM/libpyTypeConverter2.cpp]: This code has nothing to do with a function declaration. Calling `init_numpy()` without assigning it's result to anything should be OK. This change here is unnecessary.

Comment (aali1987) [reply libpyEM/libpyTypeConverter2.cpp]: Dont know why it failed in previous case.   I made the changes you suggested.

**Pull request closed without merging**